RuriLib/Blocks/Interop/Methods.cs

```diff
@@ -39,15 +40,57 @@ public static string ShellCommand(BotData data, string executable, string argume
          * These are not blocks, but they take BotData as an input. The ScriptBlockInstance will take care
          * of writing C# code that calls these methods where necessary once it's transpiled.
          */
-
-        public static async Task<T> InvokeNode<T>(BotData data, string scriptFile, object[] parameters)
+        public static async Task<T> InvokeNode<T>(BotData data, string scriptOrFile, object[] parameters, bool isScript = false, string scriptHash = null)
         {
             data.Logger.LogHeader();
-            var result = await StaticNodeJSService.InvokeFromFileAsync<T>(scriptFile, null, parameters, data.CancellationToken).ConfigureAwait(false);
+            T result;
+
+            if (isScript && !string.IsNullOrEmpty(scriptHash))
+            {
+                result = await InvokeFromStringOrCache<T>(data, scriptOrFile, parameters, scriptHash);
+            }
+            else
+            {
+                result = await InvokeFromFile<T>(data, scriptOrFile, parameters);
+            }
+
             data.Logger.Log($"Executed NodeJS script with result: {result}", LogColors.PaleChestnut);
             return result;
         }
 
+        private static async Task<T> InvokeFromStringOrCache<T>(BotData data, string script, object[] parameters, string scriptHash)
```

Comment:
`scriptHash` should be a nullable string, to match what said above. If it's null or empty, use `InvokeFromStringAsync` directly (using a null hash), otherwise try to invoke from cache like it is now.